gcp-client/src/test/java/com/google/graphite/platforms/plugin/client/ComputeClientTest.java

```diff
@@ -0,0 +1,245 @@
+
+  @InjectMocks ComputeClient computeClient;
+
+  private List<Region> listOfRegions;
+  private List<Zone> listOfZones;
+  private List<MachineType> listOfMachineTypes;
+  private List<DiskType> listOfDiskTypes;
+  private List<InstanceTemplate> listOfInstanceTemplate;
+
+  @Before
+  public void init() throws Exception {
+    listOfRegions = new ArrayList<>();
+    listOfZones = new ArrayList<>();
+    listOfMachineTypes = new ArrayList<>();
+    listOfDiskTypes = new ArrayList<>();
+    listOfInstanceTemplate = new ArrayList<>();
+
+    // Mock regions
+    RegionList regionList = new RegionList().setItems(listOfRegions);
+    Mockito.when(regionsListCall.execute()).thenReturn(regionList);
+    Mockito.when(regions.list(TEST_PROJECT_ID)).thenReturn(regionsListCall);
+    Mockito.when(compute.regions()).thenReturn(regions);
+
+    // Mock zones
+    ZoneList zoneList = new ZoneList().setItems(listOfZones);
+    Mockito.when(zonesListCall.execute()).thenReturn(zoneList);
+    Mockito.when(zones.list(TEST_PROJECT_ID)).thenReturn(zonesListCall);
+    Mockito.when(compute.zones()).thenReturn(zones);
+
+    // Mock machine types
+    MachineTypeList machineTypeList = new MachineTypeList().setItems(listOfMachineTypes);
+    Mockito.when(machineTypesListCall.execute()).thenReturn(machineTypeList);
+    Mockito.when(machineTypes.list(Mockito.anyString(), Mockito.anyString()))
+        .thenReturn(machineTypesListCall);
+    Mockito.when(compute.machineTypes()).thenReturn(machineTypes);
+
+    // Mock disk types
+    DiskTypeList diskTypeList = new DiskTypeList().setItems(listOfDiskTypes);
+    Mockito.when(diskTypesListCall.execute()).thenReturn(diskTypeList);
+    Mockito.when(diskTypes.list(Mockito.anyString(), Mockito.anyString()))
+        .thenReturn(diskTypesListCall);
+    Mockito.when(compute.diskTypes()).thenReturn(diskTypes);
+
+    // Mock instance templates
+    Compute.InstanceTemplates instanceTemplates = Mockito.mock(Compute.InstanceTemplates.class);
+    Compute.InstanceTemplates.List instanceTemplatesList =
+        Mockito.mock(Compute.InstanceTemplates.List.class);
+    Compute.InstanceTemplates.Get instanceTemplatesGet =
+        Mockito.mock(Compute.InstanceTemplates.Get.class);
+    InstanceTemplateList listOfInstanceTemplates =
+        new InstanceTemplateList().setItems(listOfInstanceTemplate);
+
+    Mockito.when(instanceTemplatesGet.execute())
+        .thenReturn(new InstanceTemplate().setName(TEST_TEMPLATE_NAME));
+    Mockito.when(instanceTemplates.get(TEST_PROJECT_ID, TEST_TEMPLATE_NAME))
+        .thenReturn(instanceTemplatesGet);
+
+    Mockito.when(instanceTemplatesList.execute()).thenReturn(listOfInstanceTemplates);
+    Mockito.when(instanceTemplates.list(TEST_PROJECT_ID)).thenReturn(instanceTemplatesList);
+    Mockito.when(compute.instanceTemplates()).thenReturn(instanceTemplates);
+  }
+
+  @Test
+  public void getRegions() throws IOException {
+    listOfRegions.clear();
```

Comment:
How difficult would it be to model the ContainerClientTest.setUpClient() approach in this class? Seems like there would be less chance for shenanigans with cleaner separation.

Reply:
Now that the mocking is simpler the conversion of existing tests shouldn't be too bad. However, since the class was lacking a lot of test coverage filling that in will be more involved. Probably enough to warrant a separate PR, given the scale of this one already.

Reply:
SGTM